tests/performance_tests/performance_test_basic.py

```diff
@@ -21,26 +21,144 @@
+        return total
+    def sum_time(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.time
+        return total
+    def sum_latency(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.latency
+        return total
+    def print_stats(self):
+        print("Chain transactions: ", self.sum_transactions())
+        print("Chain cpu: ", self.sum_cpu())
+        print("Chain net: ", self.sum_net())
+        print("Chain elapsed: ", self.sum_elapsed())
+        print("Chain time: ", self.sum_time())
+        print("Chain latency: ", self.sum_latency())
+
+class chainsContainer():
+    def __init__(self):
+        self.preData = chainData()
+        self.totalData = chainData()
+        self.startBlock = 0
+        self.ceaseBlock = 0
+    def total_transactions(self):
+        return self.totalData.sum_transactions() - self.preData.sum_transactions()
+    def total_cpu(self):
+        return self.totalData.sum_cpu() - self.preData.sum_cpu()
+    def total_net(self):
+        return self.totalData.sum_net() - self.preData.sum_net()
+    def total_elapsed(self):
+        return self.totalData.sum_elapsed() - self.preData.sum_elapsed()
+    def total_time(self):
+        return self.totalData.sum_time() - self.preData.sum_time()
+    def total_latency(self):
+        return self.totalData.sum_latency() - self.preData.sum_latency()
+    def print_stats(self):
+        print("Total transactions: ", self.total_transactions())
+        print("Total cpu: ", self.total_cpu())
+        print("Total net: ", self.total_net())
+        print("Total elapsed: ", self.total_elapsed())
+        print("Total time: ", self.total_time())
+        print("Total latency: ", self.total_latency())
+    def print_range(self):
+        print("Starting block %d ending block %d" % (self.startBlock, self.ceaseBlock))
+
+
+def waitForEmptyBlocks(node):
     emptyBlocks = 0
     while emptyBlocks < emptyBlockGoal:
-        headBlock = node1.getHeadBlockNum()
-        block = node1.processCurlCmd("chain", "get_block_info", f'{{"block_num":{headBlock}}}', silentErrors=False, exitOnError=True)
-        node1.waitForHeadToAdvance()
+        headBlock = node.getHeadBlockNum()
+        block = node.processCurlCmd("chain", "get_block_info", f'{{"block_num":{headBlock}}}', silentErrors=False, exitOnError=True)
+        node.waitForHeadToAdvance()
         if block['transaction_mroot'] == "0000000000000000000000000000000000000000000000000000000000000000":
             emptyBlocks += 1
         else:
             emptyBlocks = 0
+    return node.getHeadBlockNum()
 
-def checkTotalTrx():
-    total = 0
-    f = open("var/lib/node_00/stderr.txt")
-    stringResult = re.findall(r'trxs:\s+\d+', f.read())
-    for value in stringResult:
-        intResult = re.findall(r'\d+', value)
-        total += int(intResult[0])
+def fetchStats(total):
+    i = -1
+    f = open("var/lib/node_01/stderr.txt")
```

Comment:
Use `with open("...") as f` idiom and eliminate the subsequently redundant explicit call to `close()`.